fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostCalculator.java

```diff
@@ -106,19 +112,47 @@ public CostEstimate visitPhysicalLocalQuickSort(
             return new CostEstimate(
                     childStatistics.computeSize(),
                     statistics.computeSize(),
-                    childStatistics.computeSize());
+                    0);
         }
 
         @Override
         public CostEstimate visitPhysicalDistribute(
                 PhysicalDistribute<? extends Plan> distribute, PlanContext context) {
-            StatsDeriveResult statistics = context.getStatisticsWithCheck();
             StatsDeriveResult childStatistics = context.getChildStatistics(0);
+            DistributionSpec spec = distribute.getDistributionSpec();
+            // shuffle
+            if (spec instanceof DistributionSpecHash) {
+                return new CostEstimate(
+                        childStatistics.computeSize(),
+                        0,
+                        childStatistics.computeSize());
+            }
 
+            // replicate
+            if (spec instanceof DistributionSpecReplicated) {
+                int beNumber = ConnectContext.get().getEnv().getClusterInfo().getBackendIds(true).size();
+                int instanceNumber = ConnectContext.get().getSessionVariable().getParallelExecInstanceNum();
+                beNumber = Math.max(1, beNumber);
+
+                return new CostEstimate(
+                        childStatistics.computeSize() * beNumber,
```

Comment:
do not multiply by beNumber. In fact, for replicate mode, the beNumber is bigger, the workload is smaller.
That means we need cost of a single BE node, not the cost of cluster.

Reply:
This PR aim to get a correct plan of colocate join and bucket shuffle join. Good cost model is not including. We need to adjust it finely after this PR and #12767